From b7b6656571c4354f690fda3293af524b50cb46af Mon Sep 17 00:00:00 2001
From: Jason Lowe <jlowe@apache.org>
Date: Fri, 22 Jan 2016 21:55:01 +0000
Subject: [PATCH 1381/2848] YARN-4598. Invalid event: RESOURCE_FAILED at
 CONTAINER_CLEANEDUP_AFTER_KILL. Contributed by
 tangshangwen (cherry picked from commit
 46e5ea81e0e41646ba02391aea31481ead28aaf8)

Change-Id: I9dc8216db840074b07ee9ccada87a7fc3ca39e0e
---
 .../containermanager/container/ContainerImpl.java  |    1 +
 1 file changed, 1 insertion(+)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/container/ContainerImpl.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/container/ContainerImpl.java
index 810cbc9..3bb0241 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/container/ContainerImpl.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/container/ContainerImpl.java
@@ -314,6 +314,7 @@ public ContainerImpl(Configuration conf, Dispatcher dispatcher,
     .addTransition(ContainerState.CONTAINER_CLEANEDUP_AFTER_KILL,
         ContainerState.CONTAINER_CLEANEDUP_AFTER_KILL,
         EnumSet.of(ContainerEventType.KILL_CONTAINER,
+            ContainerEventType.RESOURCE_FAILED,
             ContainerEventType.CONTAINER_EXITED_WITH_SUCCESS,
             ContainerEventType.CONTAINER_EXITED_WITH_FAILURE))
 
-- 
1.7.9.5

